src/DFG.cpp

```diff
@@ -175,6 +175,44 @@ void DFG::combine(string t_opt0, string t_opt1) {
   }
 }
 
+void DFG::combineIter(list<string>* cPattern, list<DFGNode*>* cPatternNodes) {
+  string t_opt = cPattern->front();
+  cPattern->pop_front();
+  DFGNode* dfgNode = cPatternNodes->back();
+  for (DFGNode* succNode: *(dfgNode->getSuccNodes())) {
+    if (succNode->isOpt(t_opt) and !succNode->hasCombined()) {
+      if (cPattern->empty() == true and succNode->isSuccessorOf(cPatternNodes->front())){
+        // succNode->isSuccessorOf(cPatternNodes->front()) is not needed if the pattern is not a cycle
+        cPatternNodes->push_back(succNode);
+        DFGNode* opt0Node = cPatternNodes->front();
+        opt0Node->setCombine();
+        cPatternNodes->pop_front();
+        for(DFGNode* optNode: *cPatternNodes){
+          opt0Node->addPatternPartner(optNode);
+          optNode->setCombine();
+        }
+        return;
+      } else if(cPattern->empty() == true and !succNode->isSuccessorOf(cPatternNodes->front())){
+        continue;
+      } else{
+        cPatternNodes->push_back(succNode);
+        combineIter(cPattern, cPatternNodes);
+      }
+    }
+  }
+}
+
+void DFG::combineForIter(list<string>* cPattern, list<DFGNode*>* cPatternNodes){
```

Comment:
Can change `cPattern` to `t_targetPattern`; and change `cPatternNodes` to `t_matchedNodes`. I actually prefer having the matched nodes as returned value rather than a passed in parameter.